samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/ItemBean.java

```diff
@@ -251,7 +251,8 @@ public void setItemText(String itemText)
    */
   public double getItemScore()
   {
-    return Precision.round(itemScore, 2);
+    //return Precision.round(itemScore, 2);
+	return itemScore;
```

Comment:
What is the logic of this change? A previous developer purposely rounded the valuable down to two decimals; why are we completely losing that rounding?

Reply:
Hi

This change was done in SAM-2495 to round to 2 decimals because the user is able to enter points with too much precision when authoring (like '0.2000000029802322'). However, if you round to 2 decimals, you can lose precission as you can see by the test plan from SAM-1309 or in this example: 

1. Create an assessment with 3 questions.
2. Try to set 10 points to the assessment =>
10/3 => 3.333333... per question
3. Samigo rounds from 3.333333... to 3.33. Therefore the assessment score is 9.99 (0.33*3) => Loss of precission.

SAM-1309 fixes both SAM-2495 and SAM-1309: user is able to enter points with too much precision (like 0.2000000029802322), Samigo displays to the user 2 decimal places anywhere but Samigo works with the not rounded number (taking into account all decimal places:  0.2000000029802322) to increase precission in the assessment score calculation.

If you need more explanations feel free to ask for them. Thanks

Reply:
> user is able to enter points with too much precision when authoring (like '0.2000000029802322')

How often does an instructor *purposely* want to use more than two decimals of precision for one single question?

> Samigo displays to the user 2 decimal places anywhere but Samigo works with the not rounded number (taking into account all decimal places: 0.2000000029802322) to increase precission in the assessment score calculation.

But now you have introduced a new "bug": the calculation done by hand based on what is displayed to the user is different from the calculation done by the backend with the high-precision number.

Reply:
>How often does an instructor purposely want to use more than two decimals of precision for one single question?

It is needed every time the user has to use too many decimals on the score questions with the purpose of reaching the total assessment score that user wants to set. E.g:

Assessment score to set -> 10, number of questions -> 7, question score -> 1,428571428571429
Assessment score to set -> 10, number of questions -> 9, question score  ->1,111111111111111
Assessment score to set -> 10, number of questions -> 15, question score ->0,666666666666667
Assessment score to set-> 15, number of questions -> 19, question score ->0,789473684210526

>But now you have introduced a new "bug": the calculation done by hand based on what is displayed to the user is different from the calculation done by the backend with the high precision number.

You are right. It was done to improve the IU in order to not display too many  decimals but it introduces a new bug. 

I don't know what is the best solution. Maybe not rounding the question scores in the front end and not rounding in the back end neither. Maybe not applying this patch assuming that, in some cases (examples before), the assessment score that user wants to set can not be reached. E.g: Assessment score that user wants to set: 10, 7 questions, every question is worth 1.428571428571429. Samigo rounds it to 1.43. Assessment score will be 10.01 (1.43*7).